modules/drivers/oracle/src/metabase/driver/oracle.clj

```diff
@@ -177,13 +177,13 @@
 (defmethod sql.qp/date [:oracle :hour]             [_ _ v] (trunc :hh v))
 (defmethod sql.qp/date [:oracle :hour-of-day]      [_ _ v] (hsql/call :extract :hour (hx/->timestamp v)))
 (defmethod sql.qp/date [:oracle :day]              [_ _ v] (trunc :dd v))
-(defmethod sql.qp/date [:oracle :day-of-month]     [_ _ v] (hsql/call :extract :day v))
+(defmethod sql.qp/date [:oracle :day-of-month]     [_ _ v] (hsql/call :extract :day (hx/->timestamp v)))
```

Comment:
Perhaps we should have an `extract` function for oracle like we have for the postgres and vertica drivers? 
https://github.com/metabase/metabase/blob/7f54c619c0cc13cc7add485a1c1cebc1818317f7/src/metabase/driver/postgres.clj#L254